R/mail.R

```diff
@@ -9,40 +9,51 @@
 
 #' Creates fake mbox data
 #'
+#' Used to create a fake mbox with the specified parameters and returns the output to the console
+#'
 #' @param mlist specific mailing list associated with this mbox data
-#' @param reply_from email of sender
-#' @param reply_to email of receiver
-#' @param reply_cc carbon copy of email to additional recipients
-#' @param reply_datetime Date, time in the following format: 2023-02-11T09:30:00
-#' @param timezone Abbreviated timezone of the reply eg: HST or EST
-#' @param reply_subject subject of the email
-#' @param reply_body the body of the email
+#' @param reply_from_author name of sender. Formatted as: FIRSTNAME LASTNAME
+#' @param reply_from_email email of sender. eg: 'sender@gmail.com'
+#' @param reply_to_author name of receiver. Formatted as: FIRSTNAME LASTNAME
+#' @param reply_to_email email of receiver eg: 'receiver@gmail.com'
+#' @param reply_cc_author name of carbon copy of email to additional recipients. Formatted as: FIRSTNAME LASTNAME
+#' @param reply_cc_email email of carbon copy recipients
+#' @param reply_datetime Date, time in the following format: 2023-02-11T09:30:00  Be sure to add the capital T to separate date and time
+#' @param timezone Abbreviated timezone of the reply eg: HST
+#' @param reply_subject subject of the email as a string
+#' @param reply_body the body of the email as a string
 #' @return the content of the fake mbox you created
 #' @export
-create_fake_mbox <- function(mlist, reply_from, reply_to, reply_cc, reply_datetime, timezone, reply_subject, reply_body) {
+create_fake_mbox <- function(mlist, reply_from_author, reply_from_email, reply_to_author, reply_to_email, reply_cc_author, reply_cc_email, reply_datetime, timezone, reply_subject, reply_body) {
 
 # format the date correctly
  cdate <- format(as.POSIXct(reply_datetime, format = "%Y-%m-%dT%H:%M:%S "), "%a, %d %b %Y %H:%M:%S ")
 
+ date_time_tz <- paste0(cdate, " ", timezone)
+
+ reply_from_full_info <- paste0(reply_from_author, " ", reply_from_email)
```

Comment:
sorry, i didn't mean this is exactly the format. There are a few more minor details on the parsed_mbox from an actual dataset, no? the <>, or double quote on the parameter. Can you check the output? You should expect the user to only have to type a string, and this function does the rest. So this is very close, but you need to double check the minor differences. 

This is important because functions in Kaiaulu like the identity matching will be expecting such symbols that come with the actual raw data. If this is not clear, let me know.

Reply:
I found and implemented the correct way to format the date, time, and Tz as it comes out in the output. Here's the code I used `  cdate <- format(as.POSIXct(reply_datetime, format = "%Y-%m-%dT%H:%M:%S"), "%a, %e %b %Y %H:%M:%S %z (%Z)") `
and the output looks like this: Posting-Date: Fri, 11 Jul 2025 05:35:00 -1000 (HST)

Reply:
Yes, you're right about the <> symbols. The format seems to be FIRSTNAME LASTNAME <firstname@gmail.com>. I have edited the code to reflect this format for all information like this.

Reply:
@Rubegen sorry, I am confused. Can we specify explicitly the timezone when writing the e-mail? On the fake data generator, we must be able to specify the timezone, rather than assume it to be figured implicitly from the user computer. Otherwise, unit tests that check the timezones will be impossible to be written. 

If the fake data generator currently uses the default timezone when constructing the final field only by using the local information of the user computer, then it is impossible to create an example dataset of users collaborating in two different places, unless I force my machine into different timezones, which is impractical.

I am not seeing a way to specify the timezone via the parameter string, but I am be looking on the wrong commit. Let me know if I am misunderstanding what you did.

Reply:
> 

Yes, you're right about the <> symbols. The format seems to be FIRSTNAME LASTNAME < their email >. I have edited the code to reflect this format for all information like this.

Are you sure the First and Last name are not surrounded by double quotes in the final output of parse_mbox? Could you paste a screenshot of what you see? Maybe it is something that differs depending on the project.

Reply:
*what you see from parse_mbox of an actual dataset.

Reply:
Sorry, I was looking at a sample Mbox file by mistake, rather than the output of parse_mbox.  I outputted the data to a Json file and this is what I see. 
<img width="899" alt="Screen Shot 2023-11-17 at 10 38 04 PM" src="https://github.com/sailuh/kaiaulu/assets/96926588/6110e1c3-e652-4563-b413-c851879d8695">
It looks like all of the data is surrounded with double quotes

Reply:
@Rubegen call dt <- parse_mbox on this, and use View(dt). Does the double quote appear on the final table too?

Reply:
<img width="1221" alt="Screen Shot 2023-11-17 at 11 35 56 PM" src="https://github.com/sailuh/kaiaulu/assets/96926588/a653c1f1-2740-4c98-a185-f1e5035e2970">
 You're right, there are quotes around the names

Reply:
@Rubegen 

I see some are not (row 4), and some are (row 7). So, if there is inconsistency, my suggestion is that you do not force the double quotes, and let the user of your function decide instead, i.e. your parameter just take a `Firstname Lastname` (please add an example without double quotes on the docs). That being said, it does seem every e-mail adds <>. On your email parameter, tell the user that they can type `example@domain.org` but that your generator will incorporate the `<>`. This should suffice.

Thank you for checking!